snakemake/executors.py

```diff
@@ -2052,3 +2059,605 @@ def run_wrapper(
+        quiet=False,
+        printshellcmds=False,
+        container_image=None,
+        regions=None,
+        cache=False,
+        machine_type_prefix=None,
+        latency_wait=3,
+        local_input=None,
+        restart_times=None,
+        exec_job=None,
+        max_status_checks_per_second=1,
+    ):
+
+        # Attach variables for easy access
+        self.workflow = workflow
+        self.quiet = quiet
+        self.workdir = os.path.dirname(self.workflow.persistence.path)
+        self._save_storage_cache = cache
+        self._machine_type_prefix = machine_type_prefix
+
+        # Relative path for running on instance
+        snakefile = os.path.join(
+            self.workdir, os.path.basename(self.workflow.snakefile)
+        )
+
+        # Needs to be relative for commands run in worker (linux base)
+        self.snakefile = snakefile.replace(self.workdir + "/", "")
+        exec_job = (
+            "snakemake {target} --snakefile %s "
+            "--force -j{cores} --keep-target-files  --keep-remote "
+            "--latency-wait 0 "
+            "--attempt 1 {use_threads} "
+            "{overwrite_config} {rules} --nocolor "
+            "--notemp --no-hooks --nolock " % self.snakefile
+        )
+
+        # IMPORTANT: using Compute Engine API and not k8s == no support secrets
+        self.envvars = envvars or []
+
+        # Quit early if we can't authenticate
+        self._get_services()
+        self._get_bucket()
+
+        # Akin to Kubernetes, create a run namespace, default container image
+        self.run_namespace = str(uuid.uuid4())
+        self.container_image = container_image or "vanessa/snakemake:dev"
+        self.regions = regions or ["us-east1", "us-west1", "us-central1"]
+
+        # The project name is required, either from client or environment
+        self.project = (
+            os.environ.get("GOOGLE_CLOUD_PROJECT") or self._bucket_service.project
+        )
+        if not self.project:
+            raise WorkflowError(
+                "You must provide a --google-project or export "
+                "GOOGLE_CLOUD_PROJECT to use the Life Sciences API. "
+            )
+
+        # Keep track of build packages to clean up shutdown
+        self._build_packages = set()
+
+        super().__init__(
+            workflow,
+            dag,
+            None,
+            jobname=jobname,
+            printreason=printreason,
+            quiet=quiet,
+            printshellcmds=printshellcmds,
+            latency_wait=latency_wait,
+            restart_times=restart_times,
+            exec_job=exec_job,
+            assume_shared_fs=False,
+            max_status_checks_per_second=10,
+        )
+
+    def _get_services(self):
+        """use the Google Discovery Build to generate API clients
+           for Life Sciences, and use the google storage python client 
+           for storage.
+        """
+        from googleapiclient.discovery import build as discovery_build
```

Comment:
The first import of googleapiclient needs to be wrapped in a try statement with an error message instructing to install the missing dependency, similar to the kubernetes import in the kubernetes executor. Reason: for the snakemake-minimal package, I don't want to force people to install those dependencies if they do not need it.

Reply:
Not fixed yet?